python-mini-projects/projects/Encrypt_and_decrypt_text/aes_encode.py

```diff
@@ -0,0 +1,38 @@
+from Crypto.Cipher import AES
+from Crypto import Random
+from binascii import b2a_hex
+import sys
+
+# get the plaintext
+plain_text = sys.argv[1]
+
+# The key length must be 16 (AES-128), 24 (AES-192), or 32 (AES-256) Bytes.
+key = b'this is a 16 key'
+
+# Generate a non-repeatable key vector with a length
+# equal to the size of the AES block
+iv = Random.new().read(AES.block_size)
+
+# Use key and iv to initialize AES object, use MODE_CFB mode
+mycipher = AES.new(key, AES.MODE_CFB, iv)
+
+# Add iv (key vector) to the beginning of the encrypted ciphertext
+# and transmit it together
+ciphertext = iv + mycipher.encrypt(plain_text.encode())
+
+
+# To decrypt, use key and iv to generate a new AES object
+mydecrypt = AES.new(key, AES.MODE_CFB, ciphertext[:16])
+
+# Use the newly generated AES object to decrypt the encrypted ciphertext
+decrypttext = mydecrypt.decrypt(ciphertext[16:])
+
+# output
+file_out = open("encrypted.bin", "wb")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *file\_out*. The resource is allocated by call *builtins.open*. The resource is referenced at lines 32, 33. The resource is closed at line 33. There are other execution paths that do not contain closure statements, for example, when *BufferedWriter.write()* throws an exception. To prevent this resource leak, close *file\_out* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).